core/src/test/scala/org/apache/spark/metrics/sink/StatsdSinkSuite.scala

```diff
@@ -59,6 +61,7 @@ class StatsdSinkSuite extends SparkFunSuite {
     val props = new Properties
     defaultProps.foreach(e => props.put(e._1, e._2))
     props.put(STATSD_KEY_PORT, socket.getLocalPort.toString)
+    props.put(STATSD_KEY_REGEX, "test-[0-9]+.driver.(CodeGenerator|BlockManager)")
```

Comment:
This seems to break all the other test cases except the newly added one. Could you run this whole test suite locally and fix the bug?
```
[info] *** 4 TESTS FAILED ***
[error] Failed: Total 3673, Failed 4, Errors 0, Passed 3669, Ignored 10, Canceled 4
[error] Failed tests:
[error] 	org.apache.spark.metrics.sink.StatsdSinkSuite
```

Reply:
All of them were timing out locally, I wondered if that was to do with the statsd setup. In this case looks like the timeout is happening since the test keeps waiting for the result and filter is blocking the message from being sent. I confirmed this behaviour locally and have resolved the same. The StatsdTestSuite now  runs successfully locally.